crates/hyperswitch_connectors/src/connectors/celero/transformers.rs

```diff
@@ -125,104 +367,488 @@ impl<F, T> TryFrom<ResponseRouterData<F, CeleroPaymentsResponse, T, PaymentsResp
-    pub status_code: u16,
-    pub code: String,
-    pub message: String,
-    pub reason: Option<String>,
+    pub status: CeleroResponseStatus,
+    pub msg: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub data: Option<serde_json::Value>,
+}
+
+// Error details that can be extracted from various response fields
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct CeleroErrorDetails {
+    pub error_code: Option<String>,
+    pub error_message: String,
+    pub processor_response_code: Option<String>,
+    pub decline_reason: Option<String>,
+}
+
+impl From<CeleroErrorResponse> for CeleroErrorDetails {
+    fn from(error_response: CeleroErrorResponse) -> Self {
+        Self {
+            error_code: Some("API_ERROR".to_string()),
+            error_message: error_response.msg,
+            processor_response_code: None,
+            decline_reason: None,
+        }
+    }
+}
+
+// Function to extract error details from transaction response data
+impl CeleroErrorDetails {
+    pub fn from_transaction_response(response: &CeleroCardResponse, msg: String) -> Self {
+        // Map specific error codes based on common response patterns
+        let decline_reason = Self::map_processor_error(&response.processor_response_code, &msg);
+
+        Self {
+            error_code: None,
+            error_message: msg,
+            processor_response_code: response.processor_response_code.clone(),
+            decline_reason,
+        }
+    }
+
+    pub fn from_top_level_error(msg: String) -> Self {
+        // Map specific error codes from top-level API errors
+
+        Self {
+            error_code: None,
+            error_message: msg,
+            processor_response_code: None,
+            decline_reason: None,
+        }
+    }
+
+    /// Map processor response codes and messages to specific Hyperswitch error codes
+    fn map_processor_error(processor_code: &Option<String>, message: &str) -> Option<String> {
+        let message_lower = message.to_lowercase();
+        // Check processor response codes if available
+        if let Some(code) = processor_code {
+            match code.as_str() {
+                "05" => Some("TRANSACTION_DECLINED".to_string()),
+                "14" => Some("INVALID_CARD_DATA".to_string()),
+                "51" => Some("INSUFFICIENT_FUNDS".to_string()),
+                "54" => Some("EXPIRED_CARD".to_string()),
+                "55" => Some("INCORRECT_CVC".to_string()),
+                "61" => Some("Exceeds withdrawal amount limit".to_string()),
+                "62" => Some("TRANSACTION_DECLINED".to_string()),
+                "65" => Some("Exceeds withdrawal frequency limit".to_string()),
+                "78" => Some("INVALID_CARD_DATA".to_string()),
+                "91" => Some("PROCESSING_ERROR".to_string()),
+                "96" => Some("PROCESSING_ERROR".to_string()),
+                _ => Some("Transaction failed".to_string()),
```

Comment:
We can choose to log the code that we aren't matching, so that it would help in debugging, if required.